tilelang/autodd.py

```
@@ -0,0 +1,1156 @@
+                break
+
+            capture_out = io.StringIO()
+            capture_err = io.StringIO()
+            success = False
+            with tempfile.NamedTemporaryFile("w", suffix=".py", delete=True) as f:
+                f.write(task)
+                f.flush()
+                try:
+                    with contextlib.redirect_stdout(capture_out), contextlib.redirect_stderr(capture_err):
+                        code = compile(task, f.name, "exec")
+                        exec(code, {"__builtins__": __builtins__})
```

Ruff (0.14.10)

Use of `exec` detected

(S102)